core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlockResponse.kt

```diff
@@ -0,0 +1,102 @@
+import fr.sncf.osrd.utils.units.Length
+import fr.sncf.osrd.utils.units.Offset
+
+data class TrackRange(
+    @Json(name = "track_section") val trackSection: String,
+    var begin: Offset<TrackSection>,
+    var end: Offset<TrackSection>,
+    val direction: EdgeDirection,
+)
+
+interface PathfindingBlockResponse
+
+class PathfindingBlockSuccess(
+    // Block ids
+    val blocks: List<String>,
+
+    // Route ids
+    val routes: List<String>,
+    @Json(name = "track_section_ranges") val trackSectionRanges: List<TrackRange>,
+    val length: Length<Path>,
+
+    /** Offsets of the waypoints given as input */
+    @Json(name = "path_item_positions") val pathItemPositions: List<Offset<Path>>
+) : PathfindingBlockResponse
+
+class NotFoundInBlocks(
+    @Json(name = "track_section_ranges") val trackSectionRanges: List<TrackRange>,
+    val length: Length<Path>,
+) : PathfindingBlockResponse
+
+class NotFoundInRoutes(
+    @Json(name = "track_section_ranges") val trackSectionRanges: List<TrackRange>,
+    val length: Length<Path>,
+) : PathfindingBlockResponse
+
+class notFoundInTracks : PathfindingBlockResponse
+
+open class IncompatibleConstraint(
+    val blocks: List<String>,
+    val routes: List<String>,
+    @Json(name = "track_section_ranges") val trackSectionRanges: List<TrackRange>,
+    val length: Length<Path>,
+    @Json(name = "incompatible_ranges")
+    val incompatibleRanges: List<List<Offset<Path>>>, // List of pairs
+) : PathfindingBlockResponse
+
+class IncompatibleElectrification(
+    blocks: List<String>,
+    routes: List<String>,
+    trackSectionRanges: List<TrackRange>,
+    length: Length<Path>,
+    incompatibleRanges: List<List<Offset<Path>>> // List of pairs
+) : IncompatibleConstraint(blocks, routes, trackSectionRanges, length, incompatibleRanges)
+
+class IncompatibleLoadingGauge(
+    blocks: List<String>,
+    routes: List<String>,
+    trackSectionRanges: List<TrackRange>,
+    length: Length<Path>,
+    incompatibleRanges: List<List<Offset<Path>>> // List of pairs
+) : IncompatibleConstraint(blocks, routes, trackSectionRanges, length, incompatibleRanges)
+
+class IncompatibleSignalingSystem(
+    blocks: List<String>,
+    routes: List<String>,
+    trackSectionRanges: List<TrackRange>,
+    length: Length<Path>,
+    incompatibleRanges: List<List<Offset<Path>>> // List of pairs
+) : IncompatibleConstraint(blocks, routes, trackSectionRanges, length, incompatibleRanges)
+
+class NotEnoughPathItems : PathfindingBlockResponse
+
+val polymorphicAdapter: PolymorphicJsonAdapterFactory<PathfindingBlockResponse> =
+    PolymorphicJsonAdapterFactory.of(PathfindingBlockResponse::class.java, "status")
+        .withSubtype(PathfindingBlockSuccess::class.java, "success")
+        .withSubtype(NotFoundInBlocks::class.java, "not_found_in_blocks")
+        .withSubtype(NotFoundInRoutes::class.java, "not_found_in_routes")
+        .withSubtype(notFoundInTracks::class.java, "not_found_in_tracks")
```

Comment:
```suggestion
        .withSubtype(NotFoundInTracks::class.java, "not_found_in_tracks")
```

Also, how about `NoTrackPath` or `NoTopologicalPath` ?